### PR TITLE
fix(Core/DungeonFinder): Fixed joining to LFG dungeon if there is alr…

### LINKED ISSUE #9824: Dungeon Finder issues
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

Original report: https://github.com/chromiecraft/chromiecraft/issues/2447

Sometimes when you requeue to fill spots due to people leaving, the window notifying you that replacements have been found and to accept doesn't actually show up. You hear the sound announcing that it should be there, but it just isn't. Which leads to the accept timer running out since you can't click accept and the group not being able to get any replacements in.

I don't think it always happens, just now was the second time I've had it happen.

We were in WC. 
Healer died, released, then left group.
We requeued, everyone selected their roles, then while we waited we continued to clear.
Killed Serpentis and Verdan at the top, then three of us teleported out and back in (to get to the start), while the dungeon leader just left group instead.
The three of us still in the group requeued again, then teleported out to do stuff while we waited for the rest of the group to fill.
The sound when your RDF queue pops sounded, but nothing showed up.
The alert timed out, removing us from the queue.

https://i.imgur.com/FP4vrAH.png 

Tester:

The issue is confirmed.

If Dungeon Guide leave party while queued and new Dungeon Guide formed queue - on Dungeon Finder proc system require accept from all player included those who formed queue

Button `Yes` in DF window is available after queued in some occasions. That is lead to `Internal LFG Error` and leave queue. Look to 2nd part of video. Maybe on multiple clicking or because button `Yes` available to not Dungeon Guide.

https://user-images.githubusercontent.com/85256266/146943933-a8d3bf2d-2c46-41a6-b18f-1ab4d6f365b1.mp4

### Expected Blizzlike Behaviour

In this case there is no need accept from players who formed queue. Which is going on when there is no issue with Dungeon Guide.

Button `Yes` should not be available after queued

### Source

_No response_

### Steps to reproduce the problem

Form DF party
Get inside instance
Leave party with someone who isn't Dungeon Guide
Queue in DF system
Leave party with Dungeon Guide
Queue in DF system

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/fc16abc30ea2e9412627ddd9d15868eac8aabb53

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
…eady a LFG group inside it.iterator

Fixes #9824

## Issues Addressed:

- Closes #9824
- Closes https://github.com/chromiecraft/chromiecraft/issues/2447

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Form DF party
Get inside instance
Leave party with someone who isn't Dungeon Guide
Queue party in DF system
Leave with another one player
Queue party in DF system again
Now queue these two players outside dungeon
Now these players should properly enter the dungeon.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by UltraNix** [comment]
@azerothcore/testers Please, test it.

**Comment by pklloveyou** [comment]
Tested

File: src/server/game/DungeonFinding/LFGGroupData.cpp [hunk 1]
```diff
@@ -22,19 +22,31 @@
 namespace lfg
 {
     LfgGroupData::LfgGroupData(): m_State(LFG_STATE_NONE), m_OldState(LFG_STATE_NONE),
-        m_Dungeon(0), m_KicksLeft(sWorld->getIntConfig(CONFIG_LFG_MAX_KICK_COUNT))
+        m_Dungeon(0), m_KicksLeft(sWorld->getIntConfig(CONFIG_LFG_MAX_KICK_COUNT)), _isLFGGroup(false)
```
Comment (acidmanifesto) [inline src/server/game/DungeonFinding/LFGGroupData.cpp:25]: image

**Review by acidmanifesto — DISMISSED** [review]

File: src/server/game/DungeonFinding/LFGGroupData.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/DungeonFinding/LFGGroupData.cpp:25]: i guess re-arrange the order.

**Comment by Nyeriah** [comment]
@UltraNix could you check this one? The build looks broken

**Comment by UltraNix** [comment]
Done.

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**